File: app/src/main/java/com/stupidtree/hichat/ui/chat/FriendState.java
package com.stupidtree.hichat.ui.chat;

/**
 * 好友状态
 */
public class FriendState {
    enum STATE{ONLINE,OFFLINE,YOU,OTHER}
    STATE state;

    public static FriendState getOnline(){
        FriendState fs = new FriendState();
        fs.setState(STATE.ONLINE);
        return fs;
    }
    public static FriendState getOffline(){
        FriendState fs = new FriendState();
        fs.setState(STATE.OFFLINE);
        return fs;
    }

    public static FriendState getWithYou(){
        FriendState fs = new FriendState();
        fs.setState(STATE.YOU);
        return fs;
    }
    public static FriendState getWithOther(){
        FriendState fs = new FriendState();
        fs.setState(STATE.OTHER);
        return fs;
    }
    public STATE getState() {
        return state;
    }

    public void setState(STATE state) {
        this.state = state;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/chat/FriendState.kt
package com.stupidtree.cloudliter.ui.chat

/**
 * 好友状态
 */
class FriendState {
    enum class STATE {
        ONLINE, OFFLINE, YOU, OTHER
    }

   var state: STATE? = null


    companion object {
        val online: FriendState
            get() {
                val fs = FriendState()
                fs.state = STATE.ONLINE
                return fs
            }

        val offline: FriendState
            get() {
                val fs = FriendState()
                fs.state = STATE.OFFLINE
                return fs
            }

        val withYou: FriendState
            get() {
                val fs = FriendState()
                fs.state = STATE.YOU
                return fs
            }

        val withOther: FriendState
            get() {
                val fs = FriendState()
                fs.state = STATE.OTHER
                return fs
            }
    }
}